tgg/research/gnomad_tgps.py

```diff
@@ -0,0 +1,239 @@
+analysis. The script is a very simple variant count algorithm, so hopefully very
+straightforward to run. We are currently writing up this work, and of course,
+anybody who participates in this analysis would be included as a co-author."""
+
+import hail as hl
+from gnomad_qc.v3.resources.basics import get_gnomad_v3_vds
+from gnomad_qc.v3.resources.release import release_sites
+from gnomad_qc.v3.resources.meta import meta
+import hailtop.fs as hfs
+import argparse
+import logging
+
+NUM_OF_GNOMAD_SAMPLES = 76156
+OUTPUT_FILENAME = "pangemoe_assessment_stats.txt"
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("Pangenome_assessment")
+logger.setLevel(logging.INFO)
+
+
+def get_pangenome_ids(
+        id_path: str,
+) -> hl.expr.expressions.typed_expressions.SetExpression:
+    """
+    Generate a Hail set of sample IDs in the pangenome.
+
+    :param str id_path: Cloud path to read in the .txt file containing pangenome sample
+    IDs.
+    :return: A set of sample IDs as hl.expr.expressions.typed_expressions.SetExpression.
+    """
+    ids = []
+    with hfs.open(id_path, 'r') as file:
+        for line in file:
+            sid = line.split(' ')[1].strip()
+            ids.append(sid)
+    ids = hl.set(ids)
+    return ids
+
+
+def write_output(
+        dat: hl.utils.struct.Struct,
+        file_dir: str,
+        filename: str,
+) -> None:
+    """
+    Write the output Struct containing computed stats to a .txt file.
+
+    :param hl.utils.struct.Struct dat: Data containing computed stats.
+    :param str file_dir: Output directory.
+    :param str filename: Output filename.
+    :return: None; function writes output to the output path.
+    """
+    with hfs.open(f'{file_dir}/{filename}', 'w') as f:
+        print(dat, file=f)
+
+
+def compute_stats(
+        pangenome_ids_path: str,
+        output_dir: str,
+) -> None:
+    """
+    Compute the stats of variants per gnomAD sample for gnomAD samples not in the
+    pangenome.
+
+    :param str pangenome_ids_path: Cloud path for the text file containing pangenome
+    sample IDs.
+    :param str output_dir: Cloud path to output the result file.
+    :return: None; function writes .txt file to output directory.
+    """
+    # Get variant site frequency table.
+    gnomad_freq_ht = release_sites().ht()
+
+    # Get gnomAD v3 data in split form.
+    mt = get_gnomad_v3_vds().variant_data
+    mt = hl.experimental.sparse_split_multi(mt)
+
+    # Get gnomAD v3 metadata and filter to released samples.
+    meta_ht = meta.ht()
+    meta_ht = meta_ht.filter(meta_ht.release == True)
+
+    # Use the filtered metadata to filter gnomAD v3 data.
+    mt = mt.filter_cols(hl.is_defined(meta_ht[mt.col_key]))
+    num_of_v3_samples = mt.count_cols()
+    logger.info("The number of samples in gnomAD v3 is: ", num_of_v3_samples)
```

Comment:
this is where you could use the constant to check, something like:
```
if num_of_v3_samples != NUM_OF_GNOMAD_SAMPLES:
    raise DataException(f"Found {num_of_v3_samples} in VDS but was expecting {NUM_OF_GNOMAD_SAMPLES}; please double check!")
```
and you'd have to import DataException: https://github.com/broadinstitute/gnomad_methods/blob/main/gnomad/resources/resource_utils.py#L627